src/main/scala/com/gu/iosdeployments/Lambda.scala

```diff
@@ -39,14 +41,14 @@ object Lambda {
           case ("external-beta", Some(LiveAppBeta(_, _, _, _, "IN_BETA_TESTING"))) =>
             logger.info(s"External beta deployment for ${runningDeployment.version} is complete...")
             GitHubApi.markDeploymentAsSuccess(gitHubConfig, runningDeployment)
-          case ("external-beta", Some(build@LiveAppBeta(_, _, _, "IN_BETA_TESTING", "READY_FOR_BETA_SUBMISSION"))) =>
+          case ("external-beta", Some(build @ LiveAppBeta(_, _, _, "IN_BETA_TESTING", "READY_FOR_BETA_SUBMISSION"))) =>
             logger.info(s"External beta deployment for ${runningDeployment.version} can now be submitted for review...")
             AppStoreConnectApi.submitForBetaTesting(appStoreConnectToken, build.buildId)
-          case ("external-beta", Some(build@LiveAppBeta(_, _, _, _, "BETA_APPROVED"))) =>
+          case ("external-beta", Some(build @ LiveAppBeta(_, _, _, _, "BETA_APPROVED"))) =>
             logger.info(s"External beta deployment for ${runningDeployment.version} can now be distributed to users...")
             AppStoreConnectApi.distributeToExternalTesters(appStoreConnectToken, build.buildId, externalTesterConfig)
           case (_, None) =>
-            Try(logger.info(s"Found running beta deployment ${runningDeployment.version}, but build was not present in App Store Connect response"))
+            notPresentInAppStoreConnect(runningDeployment, gitHubConfig)
           case _ =>
```

Comment:
I think there are still scenarios where we could get 'stuck' at this stage. It's tricky to figure out which states are definite 'failure' states from [the Apple documentation](https://developer.apple.com/documentation/appstoreconnectapi/internalbetastate). I'll try to address this problem in a subsequent PR.

Reply:
FYI @aoifemcl15